axelor-account/src/main/java/com/axelor/apps/account/service/MoveLineExportServiceImpl.java

```diff
@@ -393,84 +393,27 @@ public MetaFile exportMoveLineTypeSelect1000(
       }
     }
 
-    List<MoveLine> moveLineList =
+    com.axelor.db.Query<MoveLine> moveLineQuery =
         moveLineRepo
             .all()
             .filter(moveLineQueryStr)
             .order("move.accountingDate")
             .order("date")
-            .order("name")
-            .fetch();
+            .order("name");
 
-    if (!moveLineList.isEmpty()) {
-      List<Move> moveList = new ArrayList<>();
-      for (MoveLine moveLine : moveLineList) {
-        String[] items = new String[18];
-        Move move = moveLine.getMove();
-        if (!moveList.contains(move)) {
-          moveList.add(move);
-        }
-        Journal journal = move.getJournal();
-        items[0] = journal.getCode();
-        items[1] = journal.getName();
-        items[2] = moveLine.getMove().getReference();
-        items[3] = moveLine.getDate().format(DateTimeFormatter.ofPattern(DATE_FORMAT_YYYYMMDD));
-        items[4] = moveLine.getAccount().getCode();
-        items[5] = moveLine.getAccount().getName();
-        items[6] = "";
-        items[7] = "";
-        Partner partner = moveLine.getPartner();
-        if (partner != null) {
-          items[6] =
-              moveLine.getAccount().getAccountType().getIsManageSubsidiaryAccount()
-                  ? partner.getPartnerSeq()
-                  : "";
-          items[7] =
-              moveLine.getAccount().getAccountType().getIsManageSubsidiaryAccount()
-                  ? partner.getName()
-                  : "";
-        }
-        String origin = moveLine.getOrigin();
-        items[8] = Strings.isNullOrEmpty(origin) ? "NA" : origin;
-        if (moveLine.getOriginDate() != null) {
-          items[9] =
-              moveLine.getOriginDate().format(DateTimeFormatter.ofPattern(DATE_FORMAT_YYYYMMDD));
-        }
-        items[10] = moveLine.getDescription();
-        items[11] = moveLine.getDebit().toString().replace('.', ',');
-        items[12] = moveLine.getCredit().toString().replace('.', ',');
-
-        ReconcileGroup reconcileGroup = moveLine.getReconcileGroup();
-        if (reconcileGroup != null
-            && reconcileGroup.getStatusSelect() == ReconcileGroupRepository.STATUS_FINAL) {
-          items[13] = reconcileGroup.getCode();
-          items[14] =
-              reconcileGroup
-                  .getDateOfLettering()
-                  .format(DateTimeFormatter.ofPattern(DATE_FORMAT_YYYYMMDD))
-                  .toString();
-        } else {
-          items[13] = "";
-          items[14] = "";
-        }
+    int offset = 0;
+    List<MoveLine> moveLineList;
+    List<Move> moveList = new ArrayList<>();
 
-        if (move.getAccountingDate() != null) {
-          items[15] =
-              move.getAccountingDate().format(DateTimeFormatter.ofPattern(DATE_FORMAT_YYYYMMDD));
-        }
+    while (!(moveLineList = moveLineQuery.fetch(10000, offset)).isEmpty()) {
```

Comment:
To improve performances even more, don't fetch a List<MoveLine> but just a List<Long> with MoveLines' ids.